### PR TITLE
[DE-248] DE-466 Extract components and site endpoints

### PR DESCRIPTION
**Pull request by michalpierog**
[DE-248] DE-466 Extract components endpoints
[DE-248] DE-468 Extract Site endpoints

[DE-248]: https://maxioevolution.atlassian.net/browse/DE-248?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DE-248]: https://maxioevolution.atlassian.net/browse/DE-248?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: components/schemas/Credit-Note.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    type: string
+    description: The ISO 4217 currency code (3 character string) representing the currency of the credit note amount fields.
+  memo:
+    type: string
+    description: "The memo printed on credit note, which is a description of the reason for the credit."
+  seller:
+    $ref: "./Invoice-Seller.yaml"
+    description: Information about the seller (merchant) listed on the masthead of the credit note.
+  customer:
+    $ref: "./Invoice-Customer.yaml"
```
Comment (maciej-nedza) [inline components/schemas/Credit-Note.yaml:56]: This model doesn't have `reference` field (which was only of type null... so idk if it should be there)

File: components/schemas/Credit-Note.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    items:
+      $ref: "./Invoice-Tax.yaml"
+  applications:
+    type: array
+    items:
+      $ref: "./Credit-Note-Application.yaml"
+  refunds:
+    type: array
+    items:
+      $ref: "./Invoice-Refund.yaml"
```
Comment (maciej-nedza) [inline components/schemas/Credit-Note.yaml:102]: this model has an additional field `gateway_transaction_id` which wasn't here, but I guess it's reused here to suppress duplication?

File: components/schemas/List-Components-Price-Points-Date-Field.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+title: List Components Price Points Date Field
```
Comment (maciej-nedza) [inline components/schemas/List-Components-Price-Points-Date-Field.yaml:1]: is it different than Basic-Date-Field.yaml?

File: components/schemas/List-Components-Date-Field.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+title: List Components Date Field
```
Comment (maciej-nedza) [inline components/schemas/List-Components-Date-Field.yaml:1]: is it different than Basic-Date-Field.yaml?

File: components/schemas/List-Products-Price-Points-Date-Field.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+title: List Products Price Points Date Field
```
Comment (maciej-nedza) [inline components/schemas/List-Products-Price-Points-Date-Field.yaml:1]: is it different than Basic-Date-Field.yaml?

File: reference/Chargify-API.v1.yaml [hunk 6]
```diff
@@ -11516,73 +11483,7 @@ paths:
...
-                          properties:
-                            mrr:
-                              type: string
-                            usage:
-                              type: string
-                            recurring:
-                              type: string
-                    test_mode:
-                      type: boolean
+                  $ref: "../components/schemas/List-Sale-Rep-Item.yaml"
```
Comment (maciej-nedza) [inline reference/Chargify-API.v1.yaml:11486]: Shouldn't it be extracted to list response model?

File: components/schemas/Sale-Rep-Item-Mrr-Data.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+title: Sale Rep Item Mrr Data
+type: object
+additionalProperties:
+  $ref: "./Sale-Rep-Item-Mrr.yaml"
+examples:
```
Comment (maciej-nedza) [inline components/schemas/Sale-Rep-Item-Mrr-Data.yaml:5]: How will it work? Is it compatible with the previous response?

**Review by maciej-nedza — APPROVED** [review]
Left some comments, LGTM

File: components/schemas/List-Components-Price-Points-Date-Field.yaml (hunk 3 above)
Comment (michalpierog) [reply components/schemas/List-Components-Price-Points-Date-Field.yaml:1]: No, but I decided to use different models. I think that if some new date field would be introduced in the future it would be per endpoint and not for all endpoints so it won't be a breaking change (specific model would be extended with a new value so the existing code should still compile).

File: components/schemas/List-Components-Date-Field.yaml (hunk 4 above)
Comment (michalpierog) [reply components/schemas/List-Components-Date-Field.yaml:1]: No, but I decided to use different models. I think that if some new date field would be introduced in the future it would be per endpoint and not for all endpoints so it won't be a breaking change (specific model would be extended with a new value so the existing code should still compile).

File: components/schemas/List-Products-Price-Points-Date-Field.yaml (hunk 5 above)
Comment (michalpierog) [reply components/schemas/List-Products-Price-Points-Date-Field.yaml:1]: No, but I decided to use different models. I think that if some new date field would be introduced in the future it would be per endpoint and not for all endpoints so it won't be a breaking change (specific model would be extended with a new value so the existing code should still compile).

File: components/schemas/Sale-Rep-Item-Mrr-Data.yaml (hunk 7 above)
Comment (michalpierog) [reply components/schemas/Sale-Rep-Item-Mrr-Data.yaml:5]: Yes, it's compatible.
It allows to use any key with Sale-Rep-Item-Mrr value (it should return map). Previously some keys were hardcoded but probably the response is dynamic and changes depending on dates.

File: reference/Chargify-API.v1.yaml (hunk 6 above)
Comment (michalpierog) [reply reference/Chargify-API.v1.yaml:11486]: No, operation should return List<List-Sale-Rep-Item> so the new model isn't required. 
It would be required if List-Sale-Rep-Response would wrap the array like:
```
{
"items": []
}
```

File: components/schemas/Credit-Note.yaml (hunk 1 above)
Comment (michalpierog) [reply components/schemas/Credit-Note.yaml:56]: I decided to use the same model as the Credit-Note has a lot in common with Invoice. 
We need proper tests for this as there are a lot of discrepancies, there isn't a lot of sense to use `type: "null"` because this means that the value is always null so it could be ignored at all and not serialised.

File: components/schemas/Credit-Note.yaml (hunk 2 above)
Comment (michalpierog) [reply components/schemas/Credit-Note.yaml:102]: Yes, and because gateway_transaction_id is nullable and this is a response then it should be ok. Also, by default all properties are optional so they do not need to appear in the response.
